app/Models/Building.php

```diff
@@ -143,6 +164,9 @@ public function getTagsAttribute()
     {
         $tags = $this->architects->pluck('full_name')->all();
         $tags[] = $this->location_city;
+        if (!empty($this->current_function)) {
+           $tags[] = $this->current_function;
+        }
         $tags[] = $this->years_span;
         return $tags;
```

Comment:
I'm not sure, but my guess is that more of these could be empty. So maybe something like this?
```suggestion
        $tags[] = $this->current_function;
        $tags[] = $this->years_span;
        return Arr::where($tags, fn ($tag) => !empty($tag));
```

Reply:
this is brilliant! I would just flatten the resulting array.. just in case...